content/sensu-go/5.15/reference/apikeys.md

```diff
@@ -0,0 +1,193 @@
+---
+title: "API Keys"
+description: "API keys are long-lived authentication tokens that are generated by the Sensu backend. You can provide API keys to applications that want to interact with the Sensu API. Read the reference doc to learn about API keys."
+weight: 10
+version: "5.15"
+product: "Sensu Go"
+menu: 
+  sensu-go-5.15:
+    parent: reference
+---
+
+- [API key compatibility](#api-key-compatibility)
+- [API key resource structure](#api-key-resource-structure)
+- [API key specification](#api-key-specification)
+- [Examples](#examples)
+
+API keys are long-lived authentication tokens that make it more convenient to authenticate with Sensu-adjacent applications. API keys are generated by the Sensu backend, and you can provide them to applications that want to interact with the Sensu API.
+
+Use the [APIKey API][1] to create, retrieve, update, and delete API keys.
+
+## API key compatibility
+
+API keys are 100% backward-compatible with Sensu's [authentication token system][2], but API keys do not need to be refreshed every 15 minutes like the auth tokens.
+
+When you specify an API key in a request, the system resolves it to an authentication token and continues through the regular authentication process.
+
+_NOTE: The API key resource is not compatible with [`sensuctl create`][3]._
+
+## API key resource structure
+
+{{< language-toggle >}}
+
+{{< highlight yml >}}
+type: apikeys
+api_version: core/v2
+metadata:
+  name: my-api-key
+spec:
+  key: key-id
+  username: user1
+  groups:
+  - group1
+  - group2
+  created_at: 1569874845
+{{< /highlight >}}
+
+{{< highlight json >}}
+{
+  "type": "apikeys",
+  "api_version": "core/v2",
+  "metadata" : {
+    "name": "my-api-key"
+  },
+  "spec": {
+    "key": "key-id",
+    "username": "user1",
+    "groups": [
+      "group1",
+      "group2"
+      ]
+    "created_at": 1569874845
+    }
+  }
+{{< /highlight >}}
+
+{{< /language-toggle >}}
+
+## API key specification
+
+### Top-level attributes
+
+type         | 
+-------------|------
+description  | Top-level attribute specifying the resource type. API keys should always be of type `apikeys`.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"type": "apikeys"{{< /highlight >}}
+
+api_version  | 
+-------------|------
+description  | Top-level attribute specifying the Sensu API group and version. This attribute should always be `core/v2`.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"api_version": "core/v2"{{< /highlight >}}
+
+metadata     | 
+-------------|------
+description  | Top-level collection of metadata about the API key, including the `name`. The `metadata` map is always at the top level of the check definition. This means that in `wrapped-json` and `yaml` formats, the `metadata` scope occurs outside the `spec` scope.
```

Comment:
This value is a UUID randomly generated by Sensu. The user cannot provide a name for an APIKey.